atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingKeyValueServiceForwardingTest.java

```diff
@@ -0,0 +1,294 @@
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.ImmutableCandidateCellForSweeping;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowColumnRangeIterator;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.keyvalue.impl.LocalRowColumnRangeIterator;
+import com.palantir.common.base.ClosableIterators;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class TrackingKeyValueServiceForwardingTest {
+    private static final long TIMESTAMP = 12L;
+    private static final byte[] BYTES_1 = new byte[1];
+    private static final byte[] BYTES_2 = new byte[2];
+    private static final Cell CELL_1 = Cell.create(BYTES_1, BYTES_1);
+    private static final Cell CELL_2 = Cell.create(BYTES_2, BYTES_2);
+    private static final Value VALUE_1 = Value.create(BYTES_1, TIMESTAMP);
+    private static final Value VALUE_2 = Value.create(BYTES_2, TIMESTAMP);
+
+    @Mock
+    private TableReference tableReference;
+
+    @Mock
+    private RangeRequest rangeRequest;
+
+    @Mock
+    private List<byte[]> rows;
+
+    @Mock
+    private Map<Cell, Long> timestampByCellMap;
+
+    @Mock
+    private Map<Cell, Value> valueByCellMap;
+
+    @Mock
+    private KeyValueService delegate;
+
+    private TrackingKeyValueService trackingKvs;
+
+    @Before
+    public void setUp() {
+        trackingKvs = new TrackingKeyValueServiceImpl(delegate);
+    }
+
+    @Test
+    public void getAsyncWrapsAndForwardsDelegateFutureResult() throws ExecutionException, InterruptedException {
+        when(delegate.getAsync(tableReference, timestampByCellMap)).thenReturn(Futures.immediateFuture(valueByCellMap));
+
+        assertThat(trackingKvs.getAsync(tableReference, timestampByCellMap).get())
+                .isSameAs(valueByCellMap);
+    }
+
+    @Test
+    public void getRowsForwardsDelegateResult() {
+        ColumnSelection columnSelection = mock(ColumnSelection.class);
+        when(delegate.getRows(tableReference, rows, columnSelection, TIMESTAMP)).thenReturn(valueByCellMap);
+
+        assertThat(trackingKvs.getRows(tableReference, rows, columnSelection, TIMESTAMP))
+                .isSameAs(valueByCellMap);
+    }
+
+    @Test
+    public void getRowsBatchColumnRangeTest() {
```

Comment:
this is a bad test name. _What_ part of getRowsBatchColumnRange is this testing?